packages/babel-helper-module-transforms/src/rewrite-live-references.js

```diff
@@ -171,11 +178,8 @@ const rewriteReferencesVisitor = {
 
     const localName = path.node.name;
 
-    const localBinding = path.scope.getBinding(localName);
-    const rootBinding = scope.getBinding(localName);
-
     // redeclared in this scope
-    if (rootBinding !== localBinding) return;
+    if (!isSameBinding(scope, path.scope, localName)) return;
```

Comment:
Consider add it to a scope method, such as
```js
scope.hasSameBinding(localName, otherScope)
```
because the `// redeclared in this scope` check is also used in `helper-simple-access` and `transform-modules-commonjs`.

Reply:
Then it wouldn't work with `@babel/core` <= 7.6.0. Let's defer this PR to Babel 8 since it's not urgent.

Reply:
🤔Doesn't uncommenting `!!binding` cause it fail with `@babel/core` <= 7.6.0?

> Let's defer this PR to Babel 8 since it's not urgent.

Yup.

Reply:
Also relying on `scope.hasSameBinding`, since in older versions it's undefined.